submit.py

```diff
@@ -226,36 +119,29 @@ def login_from_config(cfg):
     return login(loginurl, username, password, token)
 
 
-def submit(url_opener, submit_url, problem, language, files,
-           mainclass=None, tag=None):
+def submit(submit_url, cookies, problem, language, files, mainclass="", tag=""):
     """Make a submission.
 
     The url_opener argument is an OpenerDirector object to use (as
     returned by the login() function)
-    """
-    if mainclass is None:
-        mainclass = ""
-    if tag is None:
-        tag = ""
 
-    form = MultiPartForm()
-    form.add_field('submit', 'true')
-    form.add_field('submit_ctr', '2')
-    form.add_field('language', language)
-    form.add_field('mainclass', mainclass)
-    form.add_field('problem', problem)
-    form.add_field('tag', tag)
-    form.add_field('script', 'true')
+    Returns the requests.Result from the submission
+    """
 
-    if len(files) > 0:
-        for filename in files:
-            form.add_file('sub_file[]', os.path.basename(filename),
-                          open(filename))
+    data = dict(submit='true',
+                submit_ctr=2,
+                language=language,
+                mainclass=mainclass,
+                problem=problem,
+                tag=tag,
+                script='true')
```

Comment:
I guess I prefer it when the keys are strings since it is easier to write (a single = rather than two ' and a :) but I can change it if you like.